### PR TITLE
RATIS-1232. Change error log message in initGroups.

### PR DESCRIPTION
**Pull request by bharatviswa504**
## What changes were proposed in this pull request?

Move log level from warn to debug, if directory name is of not in the format of UUID.

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/RATIS-1232

## How was this patch tested?

Existing tests.

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java [hunk 1]
```diff
@@ -236,8 +236,10 @@ void initGroups(RaftGroup group) {
                   addGroup(RaftGroup.valueOf(groupId));
                 }
               } catch (Exception e) {
-                LOG.warn(getId() + ": Failed to initialize the group directory "
-                    + sub.getAbsolutePath() + ".  Ignoring it", e);
+                if (LOG.isDebugEnabled()) {
+                  LOG.debug(getId() + ": Failed to initialize the group directory "
+                      + sub.getAbsolutePath() + ".  Ignoring it", e);
+                }
```
Comment (szetszwo) [inline ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java:242]: How about printing an info message but not the stack trace?
```
                LOG.info("{}: The directory {} is not a group directory; ignoring it. {}",
                    getId(), sub.getAbsolutePath(), e);
```

**Review by szetszwo — COMMENTED** [review]
raft.server.storage.dir should be created for Ratis only.  Applications should just create a sub directory for Ratis.  It is not a good idea to use raft.server.storage.dir for other purposes.  If there are random items in it, Ratis is impossible to handle them correctly.

**Comment by bharatviswa504** [comment]
> raft.server.storage.dir should be created for Ratis only. Applications should just create a sub directory for Ratis. It is not a good idea to use raft.server.storage.dir for other purposes. If there are random items in it, Ratis is impossible to handle them correctly.

Yes, OM HA has mistakenly created the RatisSnapshot directory in ratis storage directory causing this issue during OM startup with a long error message. We are planning to fix this issue in Ozone under [HDDS-3378](https://issues.apache.org/jira/browse/HDDS-3378)

File: ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java (hunk 1 above)
Comment (bharatviswa504) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java:242]: Good idea, I will update PR.

Comment (bharatviswa504) [reply ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerProxy.java]: Updated as suggested.

**Review by szetszwo — APPROVED** [review]
+1 the change looks good.

**Pull request merged**